### PR TITLE
Authorized services updated docs.

### PR DESCRIPTION
**Pull request by acoumb**
## Description

- Updated content
- Adding access tokens and API keys manually
- API based authentication
- OAuth1 integration

## Type of suggestion

* [ ] Typo/grammar fix
* [ x ] Updated outdated content
* [ x ] New content
* [ ] Updates related to a new version
* [ ] Other

### CODE REVIEW

File: marketplace-and-integrations/packages/authorized-services.md
Comment (AndyButland) [inline marketplace-and-integrations/packages/authorized-services.md:39]:
```suggestion
A service can be configured to allow the manual entry of access tokens using the `CanManuallyProvideToken` setting. If this is set to `true`, a new section will be available for providing an access token.
```

File: marketplace-and-integrations/packages/authorized-services.md
Comment (AndyButland) [inline marketplace-and-integrations/packages/authorized-services.md:57]:
```suggestion
The interface provides two methods for retrieving the value of the access token or API key - `GetToken()` and `GetApiKey()`. These will return null if the token or key is not found, or the service is not configured to use the authorization method related to these objects.
```

File: marketplace-and-integrations/packages/authorized-services.md
Comment (AndyButland) [inline marketplace-and-integrations/packages/authorized-services.md:65]:
```suggestion
The primary use case for this package is when working with services that offer an OAuth2 authentication and authorization flow against an "app". The developer will need to create this "app" with the service.  By doing this, information such as the "client ID" and "client secret" can be applied to the configuration.
```

File: marketplace-and-integrations/packages/authorized-services.md
Comment (AndyButland) [inline marketplace-and-integrations/packages/authorized-services.md:67]:
```suggestion
In addition, the package supports API key based authorization flows.
```

Let's add `OAuth1` or other options as and when we get to them.

File: marketplace-and-integrations/packages/authorized-services.md
Comment (AndyButland) [inline marketplace-and-integrations/packages/authorized-services.md:90]:
```suggestion
            "AuthenticationMethod": "OAuth2|ApiKey",
```

File: marketplace-and-integrations/packages/authorized-services.md
Comment (AndyButland) [inline marketplace-and-integrations/packages/authorized-services.md:133]:
```suggestion
| AuthenticationMethod                                                   | An enum value that controls the type of authentication. `OAuth2` is the default value. Other available options are: `ApiKey`.                                                                                                                                                               | No       |                                            |
```

**Review by AndyButland — COMMENTED** [review]
Looks good to me.  Made a few inline comments mainly about just restricting the PR now for features we have prepared.

Suggest to save this as a draft and note that we'll add to it later, so we don't release it before we have the package updates ready to go.

**Comment by alina-tincas** [comment]
Thank you for reviewing @AndyButland 🙌 

In this case I agree that if you mention a "deadline" for when this should be merged, then we will not merge this until the mentioned date (instead of draft) 👍

File: marketplace-and-integrations/packages/authorized-services.md
Comment (AndyButland) [reply marketplace-and-integrations/packages/authorized-services.md:65]: We should also add here a section I've added to the readme:

```
When creating the app it will usually be necessary to configure a call back URL. You should use the following:

- For OAuth2: `/api/AuthorizedServiceResponse/HandleOAuth2IdentityResponse`
- For OAuth1: `/api/AuthorizedServiceResponse/HandleOAuth1IdentityResponse`
```

**Review by AndyButland — APPROVED** [review]
Approved from my side once the remaining comments are addressed.

It still shouldn't be published yet though until we've arranged a date for release.

Thanks.

File: marketplace-and-integrations/packages/authorized-services.md
Comment (acoumb) [reply marketplace-and-integrations/packages/authorized-services.md]: Updated

File: marketplace-and-integrations/packages/authorized-services.md
Comment (acoumb) [reply marketplace-and-integrations/packages/authorized-services.md]: Updated

File: marketplace-and-integrations/packages/authorized-services.md
Comment (acoumb) [reply marketplace-and-integrations/packages/authorized-services.md]: Updated

File: marketplace-and-integrations/packages/authorized-services.md
Comment (acoumb) [reply marketplace-and-integrations/packages/authorized-services.md]: Updated

File: marketplace-and-integrations/packages/authorized-services.md
Comment (acoumb) [reply marketplace-and-integrations/packages/authorized-services.md]: Updated

**Comment by acoumb** [comment]
Updated docs with some feedback points from a few weeks ago.

**Review by AndyButland — APPROVED** [review]

**Review by alina-tincas — COMMENTED** [review]
making few changes to fit with our styleguide

File: marketplace-and-integrations/packages/authorized-services.md
Comment (alina-tincas) [inline marketplace-and-integrations/packages/authorized-services.md:248]:
```suggestion
Depending on the configured authentication method, there are some methods that can be used to retrieve the access token or the API key:
```

**Review by alina-tincas — COMMENTED** [review]
one last change

**Comment by alina-tincas** [comment]
All looks good now! Thank you for the PR @acoumb and for reviewing @AndyButland 🙌 #h5YR 

Merging this in 😊

**Pull request merged**
